src/ontodia/diagram/linkLayer.tsx

```diff
@@ -125,11 +126,50 @@ export class LinkLayer extends Component<LinkLayerProps, {}> {
         this.routings = this.router.route(this.props.view.model);
     }
 
+    private getNestedGroups = (group: string, groups: {[id: string]: string}): {[id: string]: string} => {
```

Comment:
Why `getNestedGroups` takes `groups: { [id: string]: string }` and also returns `{ [id: string]: string }` at the same time? It should either only return or only accept collection to store result.